### PR TITLE
Draft 1 HW 1 .Rmd

### PR DESCRIPTION
**Pull request by mhusseinmidd**
This is the first draft for HW1.

### CODE REVIEW

File: HW-1.Rmd [hunk 1]
```diff
@@ -52,7 +52,31 @@ Please indicate
...
+  tally() %>% 
+  rename(num_flights_delayed = n)
+
+
+delayed_flights<-delayed_flights %>% 
+  mutate (Prop_flights_delayed = round(delayed_flights$num_flights_delayed/flights_day$num_flights_day,3))
+
+delayed_flights
+
+ggplot(data=delayed_flights, aes(x = month, y = Prop_flights_delayed)) + geom_line()
```
Comment (rudeboybert) [inline HW-1.Rmd]: definitely set the axes labels here.

File: HW-1.Rmd [hunk 2]
```diff
@@ -52,7 +52,31 @@ Please indicate
...
+  group_by(month) %>% 
+  filter(dep_delay>30) %>% 
+  tally() %>% 
+  rename(num_flights_delayed = n)
+
+
+delayed_flights<-delayed_flights %>% 
+  mutate (Prop_flights_delayed = round(delayed_flights$num_flights_delayed/flights_day$num_flights_day,3))
+
+delayed_flights
```
Comment (rudeboybert) [inline HW-1.Rmd]: A lot of the tables you're displaying in this HW are redundant relative to the graph. Keep one or the other. Remember: "less is more"

File: HW-1.Rmd [hunk 3]
```diff
@@ -65,7 +89,42 @@ Please indicate
 Some people prefer flying on older planes.  Even though they aren't as nice,
 they tend to have more room.  Which airlines should these people favor?
 
+
+Such customers should favor DL, since about 20% of its fleet is composed of airplanes manufactured before 1980.
```
Comment (rudeboybert) [inline HW-1.Rmd]: Keep in mind that yes, having a binary 1980 split does simplify things, but what are you losing in this simplification? For example, say one airline has all 1979 planes, but the other is all 1981? They're not that different in the grand scheme, but your approach will classify them completely differently!

File: HW-1.Rmd [hunk 4]
```diff
@@ -65,7 +89,42 @@ Please indicate
...
+
+Old_planes<-flights_planes %>% 
+  group_by(carrier) %>% 
+  tally() %>% 
+  mutate (Old_planes = round(num_old_planes_per_carrier$num_old_planes/num_planes_per_carrier$num_planes, 3)*100)
+Old_planes
+
+Old_planes %>%  arrange(desc(Old_planes))
+
+ggplot(Old_planes, aes(x=carrier, y=Old_planes)) + geom_bar(stat="identity")
```
Comment (rudeboybert) [inline HW-1.Rmd]: Consider reordering bars to an ordering that's more intuitive.  Check out: http://www.reed.edu/data-at-reed/resources/R/reordering_geom_bar.html

File: HW-1.Rmd [hunk 5]
```diff
@@ -78,22 +137,92 @@ they tend to have more room.  Which airlines should these people favor?
...
+
+flights_airports<-flights_airports %>% 
+  group_by(state) %>% 
+  mutate(WN_flights=sum(flight_indicator)) %>% 
+  group_by(state) %>% 
+  summarise (WN_flights_final=min (WN_flights)) %>% 
+  arrange (desc(WN_flights_final))
+flights_airports
+  
+q3b<-ggplot(data=flights_airports, aes(x=state, y=WN_flights_final))
```
Comment (rudeboybert) [inline HW-1.Rmd]: Again, "ALABAMA FIRST" isn't always the best way to show information.

File: HW-1.Rmd [hunk 6]
```diff
@@ -78,22 +137,92 @@ they tend to have more room.  Which airlines should these people favor?
...
+
+#Collapsing on flight: 
+firsts<-flights_weather %>% 
+  group_by(flight) %>% 
+  summarise(temp_first=first(temp), 
+            humidity_first=first(humidity), 
+            wind_speed_first=first(wind_speed),
+            delayed_first=first(dep_delay))
+
+q4a<-ggplot(firsts, aes(x=temp_first, y=delayed_first)) +geom_point() + geom_smooth(method="lm")
```
Comment (rudeboybert) [inline HW-1.Rmd]: Note, the few outlying high values of delay are crushing the y-scale of your plot, making it hard to see what's going on with the bulk of the points below.

File: HW-1.Rmd [hunk 7]
```diff
@@ -108,5 +237,36 @@ carrier flies to from Houston in the month of July.  Consider the `month()`
...
+total_number_flights
+
+#Impose the total number of flights per carrier on regional to create proportion:  
+prop_carriers<-left_join(prop_carriers, total_number_flights, by="carrier")
+
+prop_carriers<-prop_carriers %>% 
+  mutate(RPF=round(regional_flights/TNF,3))
+
+#Graph: 
+q5<-ggplot(data=prop_carriers, aes(x=carrier, y=RPF, fill = region))+ geom_bar(stat="identity")+ylab("Proportion of Regional Flights")
```
Comment (rudeboybert) [inline HW-1.Rmd]: Looks good. Remember to acknowledge what those grey boxes are in your write-up.

**Comment by rudeboybert** [comment]
Clean coding, which I appreciate. The way dplyr spews out tables is nasty. Consider:

* the kable() function from the knitr package to be used within a .Rmd file.
* [DT](https://rstudio.github.io/DT/) package

For example, try out

```r
library(DT)
library(knitr)
kable(iris)
datatable(iris)
```

**Comment by mhusseinmidd** [comment]
Hi Professor Kim,

Hope this finds you well. The feedback was quite helpful. Thank you.

When you have a minute, could you double check that the latest pull request with the final version of the homework made its way through to you? It is viewable here<https://github.com/Middlebury-Data-Science/HW-1/pull/11>.

In case it does not show on your end, I have attached the R script file here.

Cheers,
Mohamed

**Pull request merged**